File: NativeAPI/Python/b2_update_bucket.py
import requests
import json

api_url = ""  # Provided by b2_authorize_account
account_authorization_token = ""  # Provided by b2_authorize_account
account_id = ""  # Provided by b2_authorize_account
bucket_id = ""  # Provided by b2_list_buckets
bucket_type = ""  # The type of bucket: "allPublic" or "allPrivate"

response = requests.post(
    f"{api_url}/b2api/v4/b2_update_bucket",
    headers={"Authorization": account_authorization_token},
    json={
        "accountId": account_id,
        "bucketId": bucket_id,
        "bucketType": bucket_type
    }
)
response_data = response.json()
print(json.dumps(response_data, indent=2))

File: NativeAPI/Java/B2UpdateBucket.java
package com.backblaze.examples;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class B2UpdateBucket {
    public static void main(String[] args) {
        String apiUrl = ""; // Provided by b2_authorize_account
        String accountAuthorizationToken = ""; // Provided by b2_authorize_account
        String accountId = ""; // Provided by b2_authorize_account
        String bucketId = ""; // The ID of the bucket you want to update
        String bucketType = ""; // The type of bucket: "allPublic" or "allPrivate"

        String postParams = "{" +
                "\"accountId\":\"" + accountId + "\"," +
                "\"bucketId\":\"" + bucketId + "\"," +
                "\"bucketType\":\"" + bucketType + "\"" +
                "}";

        try (HttpClient client = HttpClient.newHttpClient()) {
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(apiUrl + "/b2api/v4/b2_update_bucket"))
                    .header("Authorization", accountAuthorizationToken)
                    .POST(HttpRequest.BodyPublishers.ofString(postParams))
                    .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            String jsonResponse = response.body();
            System.out.println(jsonResponse);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
